driver-legacy/src/main/com/mongodb/MongoClient.java

```diff
@@ -142,26 +139,98 @@
+     * {@code applyConnectionString} method on an instance of setting's builder class, building the setting, and adding it to an instance of
+     * {@link com.mongodb.MongoClientSettings.Builder}.
+     * </p>
+     *
+     * @param connectionString the connection string
+     * @see com.mongodb.MongoClientSettings.Builder#applyConnectionString(ConnectionString)
+     * @since 4.2
+     */
+    public MongoClient(final ConnectionString connectionString) {
+        this(connectionString, null);
     }
 
     /**
-     * Creates a MongoClient instance based on a (single) mongodb node.
+     * Create a new client with the given connection string.
      *
-     * @param host server to connect to in format host[:port]
+     * <p>
+     * For each of the settings classed configurable via {@link MongoClientSettings}, the connection string is applied by calling the
+     * {@code applyConnectionString} method on an instance of setting's builder class, building the setting, and adding it to an instance of
+     * {@link com.mongodb.MongoClientSettings.Builder}.
+     * </p>
+     *
+     * <p>Note: Intended for driver and library authors to associate extra driver metadata with the connections.</p>
+     *
+     * @param connectionString       the settings
+     * @param mongoDriverInformation any driver information to associate with the MongoClient
+     * @since 4.2
      */
-    public MongoClient(final String host) {
-        this(createServerAddress(host));
+    public MongoClient(final ConnectionString connectionString,
+                       @Nullable final MongoDriverInformation mongoDriverInformation) {
+        this(MongoClientSettings.builder().applyConnectionString(connectionString).build(), mongoDriverInformation);
+    }
+
+    /**
+     * Create a new client with the given client settings.
+     *
+     * @param settings the settings
+     * @since 4.2
+     */
+    public MongoClient(final MongoClientSettings settings) {
+        this(settings, null);
+    }
+
+    /**
+     * Creates a new client with the given client settings.
+     *
+     * <p>Note: Intended for driver and library authors to associate extra driver metadata with the connections.</p>
+     *
+     * @param settings               the settings
+     * @param mongoDriverInformation any driver information to associate with the MongoClient
+     * @since 4.2
+     */
+    public MongoClient(final MongoClientSettings settings, @Nullable final MongoDriverInformation mongoDriverInformation) {
+        this(settings, null, mongoDriverInformation);
+    }
+
+    private MongoClient(final MongoClientSettings settings,
+                       @Nullable final MongoClientOptions options,
+                       @Nullable final MongoDriverInformation mongoDriverInformation) {
+        delegate = new MongoClientImpl(settings, mongoDriverInformation);
+        this.options = options != null ? options : MongoClientOptions.builder(settings).build();
```

Comment:
Preserve the original options if provided, because they contain some properties that are not reflected in MongoClientSettings, e.g. `cursorFinalizerEnabled`